### PR TITLE
(#1150) Adds custom nci-heading classes

### LINKED ISSUE #1150: Story: Create NCIDS Heading Class
### Requirements
- The nci-heading class will be created
- There will be an nci-heading--display class which will be responsive. 
  - For widescreen, desktop, & tablet large: 3xl will be used
  - For tablet, mobile large, & mobile: xl will be used
- There will be nci-heading-h# for using the h1:h6 typeset mixins
  - The sizes and styles for these modifiers will be set per the created Figma
- There will be a `--label` modifier which will include a line under the heading which will span the width of the column of which the heading is contained in.
- The `--label` modifier shall function with all 6 of the different h1:h6 typeset mixins
- See the comment from Lilian from 29SEP2023 for further notes

### Notes
- The `h*` mixin uses the `$theme-h*-font-size` theme settings
- `usa-prose` uses the `@h*-typeset` mixins that add margins and other useful items

### DoR DS Tasks
- [x] Create/Update design of nci-heading for all modifiers as identified in the Requirements section. Put the solution into its own class space on Figma
- [x] Refine user story statement
- [x] Create Scenarios
- [x] Internal review of any changes to the Design solution for nci-heading and its modifiers on Figma
- [x] Update the Figma with this class and appropriate classes
- [x] Internal review of DS AC's

### DoD DS Tasks
- [ ] Develop DS class from design Figma comp
  - [ ] Update Documentation Site
    - [ ] Draft the guidance and all sections necessary for addition to the NCIDS for All Documentation Site. #1382 
  - [ ] Update Storybook Site
  - [ ] Update Example Site
- [ ] Internal review of the DS solution
    - [ ] Demo DS solution to client

### Resources:
* [Mockups](https://www.figma.com/file/3LA0fAGyGdzpBeNDCoulFf/NCIDS-Design-Kit---Master?type=design&node-id=4573-10890&mode=design&t=pzrA1ckmivpPjMLA-0)
* [Scenarios Document](https://nih.sharepoint.com/:w:/r/sites/NCI-OCPLNewCgovUXDesignSystem/Shared%20Documents/General/Details%20-%20Release%202/R2.3/Development%20Scenarios/nci-headings%20Scenarios.docx?d=wdcdc903560674989854c3172ad37d6a8&csf=1&web=1&e=O6sfie)

### Acceptance Criteria

**Example Site & Storybook**

```gherkin
Scenario: A user views a page with an nci-heading-display class on it
  Given the nci-heading-display class has been added to a page
    And the NCIDS Heading class has been created in the Design System
  Then the nci-heading-display will match the comp
  When the width of the browser is at tablet large or greater (>= 880px)
  Then the nci-heading-display will be displayed at 3xl
  When the nci-heading-display is smaller than tablet large (less than 880px)
  Then the nci-heading-display will be shown at xl
```

```gherkin
Scenario: A user views a page with the six variations of nci-heading-h# class on it
  Given the nci-heading-h1 through nci-heading-h6 class have been added to a page
    And the NCIDS Heading class has been created in the Design System
  Then the nci-heading-h1 through nci-heading-h6 class will match the comp
```

```gherkin
Scenario: A user views a page with the six variations of the nci-heading-h#--label class on it
  Given the nci-heading-h1--label through nci-heading-h6--label class have been added to a page
    And the NCIDS Heading class has been created in the Design System
  Then the nci-heading-h1--label through nci-heading-h6--label class will match the comp
    And the line under the heading will span the width of the container of the class
  When the browser width changes
  Then the line under the heading will adjust to match the width of the container of the class
```

**Documentation Site**

```gherkin
Scenario: An NCIDS for All user goes to the Documentation site for information on the NCIDS Heading class
  Given A user has access to the NCIDS Documentation Site
    And the NCIDS for All documentation site has been updated with this class
  Then the NCIDS Heading class page will exist under the path, Foundations > Typography > Headings
    And the default class will be nci-headings-h#, where h# stands for the six available sizes h1:h6
    And the variants will be nci-headings-display and nci-headings-h#--label, where h# stands for the six available sizes h1:h6
    And page for the NCIDS Heading class will be in the same style as all pages in the Foundations section
```

**Figma Community File**

```gherkin
Scenario: An NCIDS for All user goes to the NCIDS Design Kit for information on the NCIDS Heading class
  Given A user has access to the NCIDS Design Kit
    And the NCIDS for All Figma file has been updated for this class
  Then the NCIDS Heading text style will exist on the Tokens & Utilities Page
    And the NCIDS Heading text style will be available for designers to use under Text Styles > Heading Styles
```

### PR DESCRIPTION
**Pull request by dlescarbeau**
Closes #1150

### CODE REVIEW

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: packages/ncids-css/packages/nci-heading/internal/_nci-heading.scss [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		@include nci-heading('xl');
+		@include u-line-height('heading', 2);
+		@include at-media('tablet-lg') {
+			@include nci-heading('3xl');
+		}
+	}
+	/* NCI Heading H1 Styles */
+	&-h1 {
+		@include nci-heading('2xl');
+		&--label {
```
[the remark this answers is not in the document]
Comment (dlescarbeau) [reply packages/ncids-css/packages/nci-heading/internal/_nci-heading.scss:30]: So should it be `.nci-heading--label-h1` in that case? I was under the impression this class would look something like `.nci-heading-h1--label`

[the remark this answers is not in the document]
Comment (dlescarbeau) [reply packages/ncids-css/packages/nci-heading/internal/_nci-heading.scss:30]: Or should it just be multiple classes IE `.nci-heading-h1 .nci-heading--label` ? I did see that comment, I hadn't gotten around to using typeset yet as this was still a draft :)

**Review by karlikpj — APPROVED** [review]

File: packages/ncids-css/packages/nci-heading/internal/_nci-heading.scss [hunk 2]
```diff
@@ -0,0 +1,52 @@
+@use "uswds-core" as *;
+
+/* NCI Heading Mixin */
+@mixin nci-heading($headingSize) {
```
[the remark this answers is not in the document]
Comment (olitharp-nci) [reply packages/ncids-css/packages/nci-heading/internal/_nci-heading.scss:4]: use 

```scss
@include typeset-display;
@include typeset-h1;
... etc
```

as requested in the comment linked earlier

File: docs/content/foundations/typography/headings.mdx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+- Title shouldn't exceed a specified character limit.
+
+## Accessibility
+
+- Accessibility docs
+
+```html
+<section class="usa-section">
+	<div class="grid-container">
+		<h1 class="nci-heading--display">nci-heading--display</h1>
```
Comment (olitharp-nci) [inline docs/content/foundations/typography/headings.mdx:38]:
```suggestion
		<h1 class="nci-heading-display">nci-heading-display</h1>
```

**Review by bennettcc — APPROVED** [review]

**Comment by bennettcc** [comment]
Passed by product on 11/8

File: packages/ncids-css/packages/nci-heading/internal/_nci-heading.scss [hunk 4]
```diff
@@ -0,0 +1,56 @@
+@use "uswds-core" as *;
+
+/* Display Heading Style */
+.nci-heading-display {
```
Comment (olitharp-nci) [inline packages/ncids-css/packages/nci-heading/internal/_nci-heading.scss:4]: duplicated below?

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: testing/ncids-css-testing/stories/design-tokens/typography/fonts/fonts.scenarios.js [hunk 5]
```diff
@@ -65,4 +65,36 @@ module.exports = [
...
+		label: 'NCI Headings',
+		storyId: 'design-tokens-typography-fonts-role-heading--nci-heading',
+		viewports: [
+			{
+				label: 'desktop',
+				width: 1024,
+				height: 768,
+			},
+			{
+				label: 'tablet',
```
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/design-tokens/typography/fonts/fonts.scenarios.js:78]: from the scss media queries we need a test for default (mobile) and tablet-lg. I'm seeing desktop and tablet on this file being generated too. is this supposed to be mobile and are both desktop breakpoints supposed to be tablet-lg?

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Review by olitharp-nci — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Pull request merged**